specs/0007-pull-requests.md

```diff
@@ -0,0 +1,144 @@
+fashion. Google has [an entire document][small-cls] on the benefit of
+small changes.
+
+[small-cls]: https://google.github.io/eng-practices/review/developer/small-cls.html
+
+Commit messages in your pull request should be **meaningful**.  They
+should follow common community standards ([1][], [2][], [3][]),
+such as:
+
+[1]: https://www.theserverside.com/video/Follow-these-git-commit-message-guidelines
+[2]: https://reflectoring.io/meaningful-commit-messages/
+[3]: https://cbea.ms/git-commit/
+
+1. Separate subject from body with a blank line
+1. Limit the subject line to around 50 characters
+1. Capitalize the subject line
+1. Do not end the subject line with a period
+1. Use the imperative mood in the subject line
+1. Wrap the body at 72 characters
+1. Use the body to explain what and why vs. how
+
+("Use the imperative mood" means that your subject makes sense when
+preceded by the phrase "This commit will…".)
+
+If your commit addresses specific issues from an issue tracker, they
+should be called out in the commit message.
+
+Your pull request description should provide a broad summary of the
+changes included in the request. Unlike commit messages, the pull
+request description can take advantage of GitHub formatting for
+clarity (e.g., including diagrams, linking to documentation, etc).
+This is a good place to provide any background that you think would be
+useful to reviewers.
+
+### As an author
+
+- Provide reviewable pull requests
+
+  Ensure that your pull requests meet the guidelines set forth earlier
+  in this document.
+
+- Recognize that you will need to change your code
+
+  People are imperfect, and even though you've spent hours working on
+  a new feature implementation, someone is going to spot something
+  that you missed or bring a different perspective to the review that
+  may result in a request for changes.
+
+  Over time, participating in code reviews will make the code better
+  and will make you a better developer.
+
+- Respond to change requests promptly
+
+  If people are taking the time to review your code, you should
+  respond promptly to those reviews. This may mean you need to pivot
+  away from working on an exciting new feature in order to address the
+  requested changes.
+
+### As a reviewer
+
+- Recognize that pull requests take time
+
+  Even a well-written pull request will require time for the review.
+  Make pull request reviews a part of your day, rather than an
+  afterthought, so that you don't feel like you're trying to cram
+  reviews in between other work.
+
+- Respond to review requests promptly
+
+  If there is a code review, someone is waiting for your feedback,
+  possibly before they can continue with their work. Try to respond to
+  review requests in a timely fashion.
+
+- Provide constructive feedback
+
+  Don't just tell people that something is broken. Explain why you
+  believe something needs changing, and consider providing examples
+  where appropriate.
+
```

Comment:
Constructive feedback can also extend to the structure of the PR.

Reply:
I agree. Do you think that requires a change to the document?

Reply:
Upon further reflection, I think this needs to flesh out some of the general possibilities.  Like, problems with the specification/user cases, problems with the design, problems with the implementation.  It could be problems with how the code is organized or even how the PR is organized just to name a few.

Personally, I am structuring my PRs/work so that it will be as easy as possible to get them through the review process.  It's hard to describe my thought structure yet, but it is getting there.

Reply:
I'm ok with this for now.